src/libtorch.cc

```diff
@@ -1284,62 +1569,113 @@ ModelInstanceState::ReadOutputTensors(
-          TRITONSERVER_ERROR_INVALID_ARG,
-          (std::string("configuration expects datatype TYPE_") +
-           TRITONSERVER_DataTypeString(config_datatype) + " for output '" +
-           name + "', model provides TYPE_" +
-           TRITONSERVER_DataTypeString(output_dtype))
-              .c_str()));
-    }
+      // Verify output datatype matches datatype from model config
+      TRITONSERVER_DataType output_dtype =
+          ConvertTorchTypeToDataType(output_flat.scalar_type());
+      TRITONSERVER_DataType config_datatype = output_dtype_map_[name];
+      if (config_datatype != output_dtype) {
+        RETURN_IF_ERROR(TRITONSERVER_ErrorNew(
+            TRITONSERVER_ERROR_INVALID_ARG,
+            (std::string("configuration expects datatype TYPE_") +
+             TRITONSERVER_DataTypeString(config_datatype) + " for output '" +
+             name + "', model provides TYPE_" +
+             TRITONSERVER_DataTypeString(output_dtype))
+                .c_str()));
+      }
 
-    const char* output_buffer =
-        static_cast<const char*>(output_flat.data_ptr());
+      const char* output_buffer =
+          static_cast<const char*>(output_flat.data_ptr());
 
-    // Output tensors may not reside on the same device as model
-    torch::Device tensor_device = output_flat.device();
+      // Output tensors may not reside on the same device as model
+      torch::Device tensor_device = output_flat.device();
 
-    //  Set output shape
-    std::vector<int64_t> batchn_shape;
-    auto shape = output_tensors[op_index].sizes();
-    for (auto itr = shape.begin(); itr != shape.end(); itr++) {
-      batchn_shape.push_back(*itr);
-    }
+      // Get output shape
+      std::vector<int64_t> batchn_shape;
+      auto shape = output_tensors[op_index].toTensor().sizes();
+      for (auto itr = shape.begin(); itr != shape.end(); itr++) {
+        batchn_shape.push_back(*itr);
+      }
+
+      if (batchn_shape.size() == 0) {
+        return TRITONSERVER_ErrorNew(
+            TRITONSERVER_ERROR_INVALID_ARG,
+            (std::string("output '") + name +
+             "' is a scalar which is not supported.")
+                .c_str());
+      }
+
+      responder.ProcessTensor(
+          name, output_dtype, batchn_shape, output_buffer,
+          (tensor_device.type() == torch::kCPU) ? TRITONSERVER_MEMORY_CPU
+                                                : TRITONSERVER_MEMORY_GPU,
+          (tensor_device.type() == torch::kCPU) ? 0 : tensor_device.index());
+
+    } else if (output_tensors[op_index].isList()) {
+        // Custom handling for string/bytes tensor...
+
+      torch::List<torch::jit::IValue> output_list = output_tensors[op_index].toList();
+
+      // Get output shape
+      size_t list_length = output_list.size();
+      std::vector<int64_t> batchn_shape{list_length};
+      if (output_list.elementType()->kind() != c10::TypeKind::StringType) {
+        return TRITONSERVER_ErrorNew(
```

Comment:
In future would we want to support the list of integers/floating points?

Reply:
If we a separate process for supporting List of datatypes that are supported inside Torch. We would want to use the Complex Tensor Structure design that I had suggested earlier